src/partition/repart.c

```diff
@@ -6113,6 +6283,169 @@ static int fd_apparent_size(int fd, uint64_t *ret) {
         return 0;
 }
 
+static int have_fstab(Context *context) {
+        assert(context);
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                if (p->dropped)
+                        continue;
```

Comment:
This logic is duplicated in the loop below. Maybe extract this into `need_fstab_one`?